sdk/scheduler/src/main/java/com/mesosphere/sdk/config/DefaultConfigurationUpdater.java

```diff
@@ -143,6 +144,35 @@ public UpdateResult updateConfiguration(ServiceSpec candidateConfig) throws Conf
     }
 
     /**
+     * Detects whether the previous {@link ServiceSpec} set the user. If it didn't, we set it to "root"
+     * as Mesos treats a non-set user as "root".
+     *
+     * @param targetConfig The previous service spec from the config
+     */
+    private Optional<ServiceSpec> fixLastServiceSpec(Optional<ServiceSpec> targetConfig) {
```

Comment:
A more descriptive name might be helpful here, since this function only updates the user field, and that only when it's not already set.